contracts/CourtPresaleActivate.sol

```diff
@@ -49,10 +59,93 @@ contract CourtPresaleActivate is IsContract, ApproveAndCallFallBack {
         require(_token == address(presale.contributionToken()), ERROR_WRONG_TOKEN);
 
         // move tokens to this contract
-        require(ERC20(_token).safeTransferFrom(_from, address(this), _amount), ERROR_TOKEN_TRANSFER_FAILED);
+        ERC20 token = ERC20(_token);
+        require(token.safeTransferFrom(_from, address(this), _amount), ERROR_TOKEN_TRANSFER_FAILED);
 
+        _buyAndActivate(_from, _amount, _token);
+
+        // refund leftovers if any
+        _refund(token);
+    }
+
+    /**
+    * @dev This function needs a previous approval on the external token used for the contributed amount.
+    *      It will pull the approved tokens, convert them in Uniswap to the presale contribution token,
+    *      convert the obtained tokens in the presale instance, and activate the converted tokens in the
+    *      jurors registry instance of the Aragon Court.
+    * @param _token Address of the external contribution token used
+    * @param _amount Amount of contribution tokens to be converted and activated
+    * @param _minTokens Minimum amount of presale contribution tokens obtained in Uniswap
+    * @param _minEth Minimum amount of ETH obtained in Uniswap (Uniswap internally converts first to ETH and then to target token)
+    * @param _deadline Transaction deadline for Uniswap
+    */
+    function contributeExternalToken(
+        address _token,
+        uint256 _amount,
+        uint256 _minTokens,
+        uint256 _minEth,
+        uint256 _deadline
+    )
+        external
+    {
+        require(_amount > 0, ERROR_ZERO_AMOUNT);
```

Comment:
Not actually sure if/how Uniswap handles this, but we may want to revert if the `_token` is the contribution token. I don't remember any explicit checks in the solidity version, and it'd be awful if it let you do the trade but took the fees.